cli/src/cli.rs

```diff
@@ -1029,33 +1030,45 @@ fn send_and_confirm_transactions_with_spinner<T: Signers>(
 ) -> Result<(), Box<dyn error::Error>> {
     let progress_bar = new_spinner_progress_bar();
     let mut send_retries = 5;
+    let mut leader_schedule: Option<RpcLeaderSchedule> = None;
+    let mut leader_schedule_epoch = 0;
+    let send_socket = UdpSocket::bind("0.0.0.0:0").unwrap();
+
     loop {
         let mut status_retries = 15;
 
+        progress_bar.set_message("Finding leader node...");
+        let epoch_info = rpc_client.get_epoch_info_with_commitment(commitment)?;
+        if epoch_info.epoch > leader_schedule_epoch || leader_schedule.is_none() {
+            leader_schedule = rpc_client
+                .get_leader_schedule_with_commitment(Some(epoch_info.absolute_slot), commitment)?;
+            leader_schedule_epoch = epoch_info.epoch;
+        }
+        let tpu_address = get_leader_tpu(rpc_client, epoch_info.slot_index, &leader_schedule);
```

Comment:
Do we need to aim at a future slot here to improve our chances of success?

Reply:
I wondered about that. When it came up for SendTransactionService and BanksServer, I believe @mvines argued for the leader of current-slot +1, but ultimately capitulated. Glancing at validators.app, it looks like ping time to each node is 120ms or less. I'll put a timer on the TPU sends and see how long that takes. If that's less than 140ms, I think it still makes sense to target the current slot leader.

Reply:
Looks like the sends are running about 200ms against testnet, so I'll update to current-slot + 1